lbz/resource.py

```diff
@@ -71,19 +73,18 @@ def __call__(self) -> Response:
                 raise UnsupportedMethod(method=self.method)
             self.request.user = self._get_user(self.request.headers)
             endpoint: Callable = getattr(self, self._router[self.path][self.method])
-            response: Response = endpoint(**self.path_params)
-            return response
+            self.response = endpoint(**self.path_params)
         except LambdaFWException as err:
             if 500 <= err.status_code < 600:
                 logger.exception(err)
             else:
                 logger.warning(err, exc_info=is_in_debug_mode())
-            return err.get_response(self.request.context["requestId"])
+            self.response = err.get_response(self.request.context["requestId"])
         except Exception as err:  # pylint: disable=broad-except
             logger.exception(err)
-            return ServerError().get_response(self.request.context["requestId"])
-        finally:
-            self.post_request_hook()
+            self.response = ServerError().get_response(self.request.context["requestId"])
+        self._post_request_hook()
```

Comment:
You didn't create any test confirming that failure inside this function does not affect the response being delivered to the user.

This is a pretty important fix in this pull request and I cannot leave it as it is now 😉